### PR TITLE
Fix issues with nested YAML

### LINKED ISSUE #147: Dracon fails to parse pipeline yaml files with nested yaml
## Issue
When the dracon binary is passed the `--pipelione` flag it reads the yaml files contained in that directory. It attempts to parse the yaml files splitting objects on triple dashes (`---`). However it incorrectly splits files when there's yaml objects embedded within a yaml object. For example the following configmap containing a yaml file:

```yaml
apiVersion: v1
data:
  jira-consumer-config.yaml: |
    ---
    defaultValues:
      project: PROJ
      issueType: Vulnerability
```

This results in the following error:
```
2023/01/12 14:52:48 Failed to apply templates :---
apiVersion: v1
data:
  jira-consumer-config.yaml: ""
```

I suspect [pkg/template/load.go line 54](https://github.com/thought-machine/dracon/blob/master/pkg/template/load.go#L54) is the cause of the problem
 
### Expected behaviour
dracon should only be splitting yaml files when the object separator (`---`) is at the start of the line.

### Actual behaviour
dracon splits files when in any abratriy location in the yaml there's tripple dash. For example the following line incorrectly is considered by dracon to be the end of a yaml object:

```yaml
a: asdasd---asdasd
```

### Steps to reproduce
```bash
mkdir /tmp/tmp.dmjjR9N78W
echo "apiVersion: v1
data:
  jira-consumer-config.yaml: |
    ---
    defaultValues:
      project: PROJ
      issueType: Vulnerability
" > /tmp/tmp.dmjjR9N78W/jira-consumer.yaml

dracon --pipeline "/tmp/tmp.dmjjR9N78W" setup
```

### PR DESCRIPTION
**Pull request by mhbardsley**
Closes #147.

### CODE REVIEW

**Review by hanawatson — APPROVED** [review]
LGTM!

**Review by chrisnovakovic — CHANGES_REQUESTED** [review]
Could we also get some regression tests for this? Nothing complex, just ones covering the cases we've run into:

* a simple YAML file with no leading document separator
* a simple YAML file with a leading document separator
* a YAML file with no leading document separator but a string value containing `---`
* a YAML file with both a leading document separator and a string value containing `---`

**Comment by mhbardsley** [comment]
The tests pass locally - the pipeline here seems to fail to build Dracon, which seems to be a separate issue, not introduced by this PR.

File: pkg/template/load_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	}{
+		{
+			desc: "a simple YAML file with no leading document separator",
+			inBytes: []byte(`a: Easy
+b:
+    c: 2
+    d:
+        - 3
+        - 4
+`),
```
Comment (chrisnovakovic) [inline pkg/template/load_test.go:24]:
```suggestion
			desc: "a simple YAML file with no leading document separator",
			inBytes: []byte(heredoc.Doc(`
				a: Easy
				b:
				  c: 2
				  d:
				    - 3
				    - 4
			`)),
```

Using [github.com/makenowjust/heredoc](https://github.com/makenowjust/heredoc) is a nice way to format blocks of text in test tables like this - it keeps the input reasonably indented with the rest of the test case and (in this particular example) means there isn't a dangling line at the start of the raw string, which is easy to miss.

Comment (mhbardsley) [reply pkg/template/load_test.go:24]: Ah nice thanks! Good to know. Will update accordingly :)

**Pull request merged**